Task 2

ZavialovaAV opened the pull request.

ZavialovaAV committed:
Add files via upload

ZavialovaAV commented:
Даны две стороны треугольника и угол между ними. Определить третью сторону, площадь треугольника и радиус описанной окружности.

BalakinaEP commented on task 2.cpp:
```diff
@@ -0,0 +1,57 @@
+ * @return результат вычисления результата 
+ */
+double calculateThirdSide(double side1, double side2, double angle) {
+return sqrt(side1 * side1 + side2 * side2 - 2 * side1 * side2 * cos(angle));
```
реализация после main

BalakinaEP commented on task 2.cpp:
```diff
@@ -0,0 +1,57 @@
+ * @return результат вычисления результата 
+ */
+double calculateTriangleArea(double side1, double side2, double angle) {
+double thirdSide = calculateThirdSide(side1, side2, angle);
+return 0.5 * side1 * side2 * sin(angle);
```
то же

BalakinaEP commented on task 2.cpp:
```diff
@@ -0,0 +1,57 @@
+ * @return результат вычисления результата 
+ */
+double calculateCircumcircleRadius(double side1, double side2, double angle) {
+double thirdSide = calculateThirdSide(side1, side2, angle);
+return (side1 * side2 * thirdSide) / (4 * sqrt((side1 + side2 + thirdSide) * (side1 + side2 - thirdSide) * (side2 + thirdSide - side1) * (side1 + thirdSide - side2)));
+}
```
то же

BalakinaEP commented on task 2.cpp:
```diff
@@ -0,0 +1,57 @@
+int main() {
+double side1, side2, angle;
+cout << "Введите длину первой стороны треугольника: ";
+cin >> side1;
+cout << "Введите длину второй стороны треугольника: ";
+cin >> side2;
+cout << "Введите величину угла между сторонами в радианах: ";
+cin >> angle;
```
не проверяете ввод на возможную ошибку

BalakinaEP commented on task 2.cpp:
```diff
@@ -0,0 +1,57 @@
+cin >> side2;
+cout << "Введите величину угла между сторонами в радианах: ";
+cin >> angle;
+double thirdSide = calculateThirdSide(side1, side2, angle);
+double area = calculateTriangleArea(side1, side2, angle);
+double radius = calculateCircumcircleRadius(side1, side2, angle);
```
Убрать лишние переменные. Сразу вызывать функции

BalakinaEP commented on task 2.cpp:
```diff
@@ -0,0 +1,57 @@
+ * @param angle - угол в треугольнике
+ * @return результат вычисления результата 
+ */
+double calculateCircumcircleRadius(double side1, double side2, double angle) {
```
double calculateCircumcircleRadius(const double side1, const  double side2, const  double angle)

ZavialovaAV committed:
Add files via upload

Даны две стороны треугольника и угол между ними. Определить третью сторону, площадь треугольника и радиус описанной окружности.

BalakinaEP commented on main (2).cpp:
```diff
@@ -0,0 +1,88 @@
+{
+    double side1, side2, angle;
+    cout << "Введите длину первой стороны треугольника: ";
+    cin >> side1;
+    cout << "Введите длину второй стороны треугольника: ";
+    cin >> side2;
+    cout << "Введите величину угла между сторонами в радианах: ";
+    cin >> angle;
```
не проверили ввод, не проверили, что введенные значения положительные

ZavialovaAV committed:
Add files via upload

BalakinaEP commented on main.cpp:
```diff
@@ -0,0 +1,101 @@
+{ 
+ 
+ cout << "Введите длину первой стороны треугольника: "; 
+ double side1 = getValue(); 
+ cout << "Введите длину второй стороны треугольника: "; 
+ double side2 = getValue(); 
+ cout << "Введите величину угла между сторонами в радианах: "; 
+ double angle = getValue(), convertRadiansToDegrees(double radians);
```
нерабочий код

BalakinaEP replied:
не проверили, что введенные значения положительные, угол от 0 до 180

ZavialovaAV committed:
Add files via upload

ZavialovaAV committed:
Update main.cpp

ZavialovaAV committed:
Add files via upload

Вычислить расстояние между двумя точками с заданными координатами.

ZavialovaAV committed:
Add files via upload

ZavialovaAV committed:
Delete main (2).cpp
